src/java/main/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1236,7 +1237,7 @@ public void run() {
                                     "Session 0x"
                                             + Long.toHexString(getSessionId())
                                             + " for server "
-                                            + clientCnxnSocket.getRemoteSocketAddress()
+                                            + serverAddress
                                             + ", unexpected error"
                                             + RETRY_CONN_MSG, e);
```

Comment:
I think it would be good to address the other issue Paul mentioned - no need to dump a stack if we know this is NoRouteToHostException - why wouldn't we add another elseif to check for this type?

Reply:
We're talking about the same thing in the Jira. It's arguable which exception should be logged at INFO level without the stack trace, I ended up separating SocketExceptions altogether and leaving the rest for the original handler.